src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/Generated/Recipe.cs

```diff
@@ -330,18 +332,94 @@ private void ConfigureBeanstalkEnvironment(Configuration settings)
                 }
             }
 
-            if (!settings.BeanstalkEnvironment.CreateNew)
-                throw new InvalidOrMissingConfigurationException("The ability to deploy an Elastic Beanstalk application to an existing environment via a new CloudFormation stack is not supported yet.");
+            if (settings.BeanstalkEnvironment.CreateNew)
+            {
+                BeanstalkEnvironment = new CfnEnvironment(this, nameof(BeanstalkEnvironment), InvokeCustomizeCDKPropsEvent(nameof(BeanstalkEnvironment), this, new CfnEnvironmentProps
+                {
+                    EnvironmentName = settings.BeanstalkEnvironment.EnvironmentName,
+                    ApplicationName = settings.BeanstalkApplication.ApplicationName,
+                    PlatformArn = settings.ElasticBeanstalkPlatformArn,
+                    OptionSettings = optionSettingProperties.ToArray(),
+                    CnamePrefix = !string.IsNullOrEmpty(settings.CNamePrefix) ? settings.CNamePrefix : null,
+                    // This line is critical - reference the label created in this same stack
+                    VersionLabel = ApplicationVersion.Ref,
+                }));
+            }
+            else
+            {
+                UpdateBeanStalkEvironment(settings);
+            }
+        }
 
-            BeanstalkEnvironment = new CfnEnvironment(this, nameof(BeanstalkEnvironment), InvokeCustomizeCDKPropsEvent(nameof(BeanstalkEnvironment), this, new CfnEnvironmentProps
+        private void UpdateBeanStalkEvironment(Configuration settings)
+        {
+            Provider customResourceProvider = new Provider(this, "CustomResourceProvider", new ProviderProps
             {
-                EnvironmentName = settings.BeanstalkEnvironment.EnvironmentName,
-                ApplicationName = settings.BeanstalkApplication.ApplicationName,
-                PlatformArn = settings.ElasticBeanstalkPlatformArn,
-                OptionSettings = optionSettingProperties.ToArray(),
-                CnamePrefix = !string.IsNullOrEmpty(settings.CNamePrefix) ? settings.CNamePrefix : null,
-                // This line is critical - reference the label created in this same stack
-                VersionLabel = ApplicationVersion.Ref,
+                OnEventHandler = GetOnEventHandler(),
+                IsCompleteHandler = GetIsCompleteHandler()
+            });
+
+            new CustomResource(this, "CustomResource", new CustomResourceProps
```

Comment:
Rename "CustomResource" to "EnvironmentUpdateResource". We might want other custom resources in the future so we don't want to cause any name conflicts. Plus I think it is better for the final stack to have the resources have reasonable logical names.

Reply:
Fixed